### PR TITLE
fix: Added port information check in ClusterIPAddress

### PR DESCRIPTION
**Pull request by ithesadson**
Signed-off-by: ithesadson <thesadson@gmail.com>

Since the change I made before was a breaking change, I just fix the mistake and send a pr again.
Issue : #4110
Previous PR: #4079

If the clusterIPAddress contains port information, checking if the port information is entered in the clusterIPAddress will cause some errors.

Error example in current code: User enters the following values.
clusterIPAddress: "https://cassandra.test/"
port: "9042"
->Expected clusterIPAddress : https://cassandra.test:9042/
The actual clusterIPAddress: https://cassandra.test/

### Checklist
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ Here is an overview of all new **experimental** features:
 - **CPU Memory Scaler** Store forgotten logger ([#4022](https://github.com/kedacore/keda/issues/4022))
   **Kafka Scaler**: Support 0 in activationLagThreshold configuration ([#4137](https://github.com/kedacore/keda/issues/4137))
 - **Prometheus Metrics**: Expose Prometheus Metrics also when getting ScaledObject state ([#4075](https://github.com/kedacore/keda/issues/4075))
+- **Cassandra Scaler**: Checking whether the port information is entered in the ClusterIPAddres is done correctly. ([#4110](https://github.com/kedacore/keda/issues/4110))
```
Comment (zroubalik) [inline CHANGELOG.md:77]: nit, please sort alphabetically

**Review by zroubalik — APPROVED** [review]
LGTM, thanks for the contribution!

**Review by zroubalik — CHANGES_REQUESTED** [review]
One more thing, do you think you can add unit test for this fix? https://github.com/kedacore/keda/blob/main/pkg/scalers/cassandra_scaler_test.go

**Comment by ithesadson** [comment]
I added the unit test. But e2e-tests has been running for about 5 days and has not finished yet. Is there any problem with my unit test?

**Comment by zroubalik** [comment]
@ithesadson maintainers need to start e2e tests, I've just triggered them

**Review by zroubalik — APPROVED** [review]
LGTM, thanks for the fix!

**Pull request merged**